python/ray/dashboard/state_aggregator.py

```diff
@@ -361,20 +363,28 @@ async def list_objects(self, *, option: ListApiOptions) -> ListApiResponse:
             {object_id -> object_data_in_dict}
             object_data_in_dict's schema is in ObjectState
         """
-        raylet_ids = self._client.get_all_registered_raylet_ids()
+        all_node_info_reply = await self._client.get_all_node_info(
+            timeout=option.timeout, limit=None
+        )
```

Comment:
we should filter ALIVE nodes, like line 485?

Reply:
Actually `get_all_node_info` accepts a `filters` argument so we can let GCS only returns ALIVE nodes.

Reply:
Addressed in https://github.com/ray-project/ray/pull/50605/commits/daa21d8d6b0330bf855007406da41709459ed452.

I also found that the filter does not support filter `node_info.runtime_env_agent_port is not None`. So for this one I still use list comprehension.